File: src/test/java/com/example/screenplay/question/QuestionWithDefaultSubject.java
package com.example.screenplay.question;

import net.serenitybdd.screenplay.Question;

public abstract class QuestionWithDefaultSubject<T> implements Question<T> {

  @Override
  public String toString() {
    return getSubject();
  }

  @Override
  public String getSubject() {
    return getClass()
        .getSimpleName()
        .replaceAll("([a-z])([A-Z])", "$1 $2");
  }
}

File: src/test/kotlin/com/example/screenplay/question/QuestionWithDefaultSubject.kt
package com.example.screenplay.question

import net.serenitybdd.screenplay.Question

abstract class QuestionWithDefaultSubject<T> : Question<T> {
  override fun getSubject(): String {
    return this::class.simpleName!!.replace("([a-z])([A-Z])".toRegex(), "$1 $2")
  }

  override fun toString(): String {
    return subject
  }
}